config/config.go

```diff
@@ -301,6 +301,12 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 			}
 		}
 
+		if scfg.BodySizeLimit == 0 {
+			if c.GlobalConfig.BodySizeLimit > 0 {
```

Comment:
I don't think we need this particular `if` condition, i.e. `>0` for the global limit.